google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystem.java

```diff
@@ -145,7 +148,35 @@ public static CloudStorageFileSystem forBucket(String bucket, CloudStorageConfig
         !bucket.startsWith(URI_SCHEME + ":"), "Bucket name must not have schema: %s", bucket);
     checkNotNull(config);
     return new CloudStorageFileSystem(
-        new CloudStorageFileSystemProvider(config.userProject()), bucket, config);
+        getCloudStorageFileSystemProvider(config, null), bucket, config);
+  }
+
+  /**
+   * Creates new file system provider instance for {@code CloudStorageFileSystemProvider}.
+   *
+   * @param config CloudStorageConfiguration to initialize {@code CloudStorageFileSystemProvider}
+   *     object
+   * @param storageOptions
+   * @return {@code CloudStorageFileSystemProvider} object
+   */
+  private static CloudStorageFileSystemProvider getCloudStorageFileSystemProvider(
+      CloudStorageConfiguration config, StorageOptions storageOptions) {
+    CloudStorageFileSystemProvider cloudStorageFileSystemProvider =
+        (storageOptions == null)
+            ? new CloudStorageFileSystemProvider(config.userProject())
+            : new CloudStorageFileSystemProvider(config.userProject(), storageOptions);
+    Set<CloudStorageFileSystemProvider> cloudStorageFileSystemProviderSet = new HashSet<>();
```

Comment:
too long names for local variables makes hard to read
I recommend: 
  cloudStorageFileSystemProvider  -> newProvider
  cloudStorageFileSystemProviderSet  -> existingProviders

Reply:
Done